pandas/_libs/tslibs/parsing.pyx

```diff
@@ -760,15 +780,20 @@ def _format_is_iso(f) -> bint:
     return False
 
 
-def _guess_datetime_format(dt_str, dayfirst=False, dt_str_parse=du_parse,
-                           dt_str_split=_DATEUTIL_LEXER_SPLIT):
+def _guess_datetime_format(
+    dt_str,
```

Comment:
str?

Reply:
Python, raises on tests with this change.